src/schemes.cpp

```diff
@@ -0,0 +1,555 @@
+// Copyright 2018 Chia Network Inc
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <cstring>
+#include <set>
+#include <string>
+
+#include "bls.hpp"
+#include "elements.hpp"
+#include "schemes.hpp"
+// #include "signature.hpp"
+
+using std::string;
+using std::vector;
+
+namespace bls {
+/* These are all for the min-pubkey-size variant.
+   TODO : analogs for min-signature-size
+*/
+
+const uint8_t *BasicScheme::CIPHERSUITE_ID =
```

Comment:
Can you put a comment on each one? And also a link to the spec
https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-02